core/src/main/java/org/apache/accumulo/core/conf/SiteConfiguration.java

```diff
@@ -82,34 +80,49 @@ public SiteConfiguration(URL accumuloPropsLocation) {
   @SuppressFBWarnings(value = "URLCONNECTION_SSRF_FD",
       justification = "location of props is specified by an admin")
   public SiteConfiguration(URL accumuloPropsLocation, Map<String,String> overrides) {
-    this.overrides = overrides;
+    config = createMap(accumuloPropsLocation, overrides);
+  }
 
-    init();
-    PropertiesConfiguration config = new PropertiesConfiguration();
+  private static ImmutableMap<String,String> createMap(URL accumuloPropsLocation,
+      Map<String,String> overrides) {
+    CompositeConfiguration config = new CompositeConfiguration();
+    config.setThrowExceptionOnMissing(false);
     config.setDelimiterParsingDisabled(true);
+    PropertiesConfiguration propsConfig = new PropertiesConfiguration();
+    propsConfig.setDelimiterParsingDisabled(true);
     if (accumuloPropsLocation != null) {
       try {
-        config.load(accumuloPropsLocation.openStream());
+        propsConfig.load(accumuloPropsLocation.openStream());
       } catch (IOException | ConfigurationException e) {
         throw new IllegalArgumentException(e);
       }
     }
-    internalConfig.addConfiguration(config);
+    config.addConfiguration(propsConfig);
+
+    // Add all properties in config file
+    Map<String,String> result = new HashMap<>();
+    config.getKeys().forEachRemaining(key -> result.put(key, config.getString(key)));
 
-    Map<String,String> temp = StreamSupport
-        .stream(((Iterable<String>) internalConfig::getKeys).spliterator(), false)
-        .collect(Collectors.toMap(Function.identity(), internalConfig::getString));
+    // Add all overrides
+    overrides.keySet().forEach(key -> result.put(key, overrides.get(key)));
```

Comment:
Not sure, but may be able to do the following :

```java
overrides.forEach(result::put);
```

Reply:
I did this